test/e2e/azure_test.go

```diff
@@ -540,4 +540,56 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating a dual-stack cluster [REQUIRED]", func() {
```

Comment:
Set it as `[REQUIRED]` because we're going to use CAPZ for k/k dual-stack tests and it'll be good to have it tested with every PR

Reply:
is it necessary to test both IPv6 and dual stack on every PR? Trying to keep the number of full cluster buildouts limited so we can keep the presubmit prow job duration as low as possible

Reply:
If we only want to test one of it, I would prefer we test dual-stack. Dual-stack config will be more commonly used compared to single-stack IPv6.

Are optional tests run periodically?

Reply:
yes the full suite of tests runs periodically https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-e2e-full-main

Reply:
perfect! i'll move dual-stack to `[OPTIONAL]` for now as it's a new test. We can decide at a later if we want to make this required and make ipv6 optional.

Based on the test run from last commit, we know the dual-stack ci flavor + tests work, so that's a good sign!